### DISCUSSION TITLE
A+B Option?

### DISCUSSION THREAD

**Discussion by MattGamingV1** [bloc97/Anime4K#234] [Ideas]
I just realized I can do 2 filters at once. I like A the most and A+A is ALMOST perfect its just a TINY bit too sharp. but theres no option for A+B only A+A B+B and A+C is there a reason its not a feature or do I not have it setup in the keybindings?

**Comment by yooooooLeon** [comment]
您的邮件已收到！谢谢！

**Comment by Tama47** [comment]
> is there a reason its not a feature or do I not have it setup in the keybindings?

The shaders are meant to be chained up together per your preferences. Mode A, B, and C are just presets that Bloc has provided.

<details>
<summary>See the <a href="https://github.com/bloc97/Anime4K/blob/master/md/GLSL_Instructions_Advanced.md">Advanced Instructions</a></summary>

Overview of default modes:
| Mode         | Shaders |
| ------------- |-------------|
| A | `Restore -> Upscale -> Upscale` |
| B | `Restore_Soft -> Upscale -> Upscale` |
| C | `Upscale_Denoise -> Upscale` |
| A+A | `Restore -> Upscale -> Restore -> Upscale` |
| B+B | `Restore_Soft -> Upscale -> Restore_Soft -> Upscale` |
| C+A | `Upscale_Denoise -> Restore -> Upscale` |

How the modes are defined:

  -  Mode A is defined initially as: `Restore -> Upscale`
  -  Mode B is defined initially as: `Restore_Soft -> Upscale`
  -  Mode C is defined initially as: `Upscale`
  -  If the mode does not start with a `Restore` shader, it must start with a `Upscale_Denoise` or `Denoise` shader, as almost every video compression algorithm is lossy.
  -  All modes have to add upscale shaders until the entire shader pipeline upscales at least 4x. A reasonable assumption is the smallest reasonable video size being 480p and the largest screen being 4K, upscaling at 4x is close to the 4.5x of 480p->4K.

</details>
